src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -72,14 +72,24 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       response.sendRedirect("/index.html");
       return;
     }
-
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
     String recipient = request.getParameter("recipient");
 
-    Message message = new Message(user, text, recipient);
+    String regex = "(https?://\\S+\\.(png|jpg))";
+    String replacement = "<img src=\"$1\" />";
+    String textWithImagesReplaced = userText.replaceAll(regex, replacement);
+
+    Message message = new Message(user, textWithImagesReplaced, recipient);
     datastore.storeMessage(message);
 
+    // String user = userService.getCurrentUser().getEmail();
+    // String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+    //
+
+    // Message message = new Message(user, text, recipient);
+    // datastore.storeMessage(message);
```

Comment:
Delete this.  It's archived in Git, so you can always get it back if you need it.